src/KubernetesClientConfiguration.cs

```diff
@@ -97,79 +92,64 @@ public KubernetesClientConfiguration(FileInfo kubeconfig = null, string currentC
-
-                if (activeContext == null)
-                {
-                    throw new KubeConfigException($"CurrentContext: {currentContext} not found in contexts in kubeconfig");
-                }
+                throw new KubeConfigException($"No users found in kubeconfig");
+            }
 
-                this.CurrentContext = activeContext.Name;
+            // current context
+            currentContext = currentContext ?? k8SConfig.CurrentContext;
+            Context activeContext = k8SConfig.Contexts.FirstOrDefault(c => c.Name.Equals(currentContext, StringComparison.OrdinalIgnoreCase));
+            if (activeContext == null)
+            {
+                throw new KubeConfigException($"CurrentContext: {currentContext} not found in contexts in kubeconfig");
             }
 
-            if (k8SConfig.Clusters == null)
+            this.CurrentContext = activeContext.Name;
+
+            // cluster
+            var cluster = k8SConfig.Clusters.FirstOrDefault(c => c.Name.Equals(activeContext.ContextDetails.Cluster, StringComparison.OrdinalIgnoreCase));
+            if (cluster?.ClusterEndpoint == null)
             {
-                throw new KubeConfigException($"clusters not found for current-context :{activeContext} in kubeconfig");
+                throw new KubeConfigException($"Cluster not found for context {activeContext} in kubeconfig");
             }
 
-            var clusterDetails = k8SConfig.Clusters.FirstOrDefault(c => c.Name.Equals(activeContext.ContextDetails.Cluster, StringComparison.OrdinalIgnoreCase));
-            if (clusterDetails?.ClusterEndpoint != null)
+            if (string.IsNullOrWhiteSpace(cluster.ClusterEndpoint.Server))
             {
-                if (string.IsNullOrWhiteSpace(clusterDetails.ClusterEndpoint.Server))
-                {
-                    throw new KubeConfigException($"server not found for current-context :{activeContext} in kubeconfig");
-                }
-
-                if (!clusterDetails.ClusterEndpoint.SkipTlsVerify &&
-                    string.IsNullOrWhiteSpace(clusterDetails.ClusterEndpoint.CertificateAuthorityData))
-                {
-                    throw new KubeConfigException($"certificate-authority-data not found for current-context :{activeContext} in kubeconfig");
-                }
-
-                this.Host = clusterDetails.ClusterEndpoint.Server;
-                this.SslCaCert = clusterDetails.ClusterEndpoint.CertificateAuthorityData;
-                this.SkipTlsVerify = clusterDetails.ClusterEndpoint.SkipTlsVerify;
+                throw new KubeConfigException($"Server not found for context {activeContext} in kubeconfig");
             }
-            else
+
+            if (!cluster.ClusterEndpoint.SkipTlsVerify &&
+                string.IsNullOrWhiteSpace(cluster.ClusterEndpoint.CertificateAuthorityData))

```

Comment:
Certificate Authority is not actually required (imagine for example that a user uses a certificate signed by a generally accepted authority [e.g. let's encrypt]) it will all just work because of the default roots.